From 40a12554da351405b6912fc360d4f9022916b17a Mon Sep 17 00:00:00 2001
From: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Date: Sun, 12 Aug 2018 10:33:18 +0530
Subject: [PATCH 6/6] bluetooth: Fix for Can't enter "Connected devices" after
 pairing

Issue: Once the device is paired and connected the Connected devices
app is crashing.

Reason: When the Setting App is checking for the connected devices to
be updated on the UI, it is checking for the type of Audio profile i.e.
whether the profile is HEADSET or A2DP. Here we are getting the profile
as A2DP i.e. A2DP source profile And then checking for the A2DP
connection status. But in this case, BXT is acting as A2DP Sink device,
so its not able to find the reference for A2DP source profile, due to
which app is crashing.

Fix: Added a check for the null case, where it is checking if no
reference for A2DP source profile is found then don't check for its
connection status. Also, if no A2DP source profile is found, check if
A2DP sink profile is present. If present, then return the connection
status for A2DP sink profile otherwise return false.

Tracked-On: None

Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 .../bluetooth/CachedBluetoothDevice.java           | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java b/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
index bb8fbe2d..4e124eaf 100644
--- a/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
+++ b/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
@@ -1189,8 +1189,18 @@ public class CachedBluetoothDevice implements Comparable<CachedBluetoothDevice>
      * @return {@code true} if {@code cachedBluetoothDevice} is a2dp device
      */
     public boolean isA2dpDevice() {
-        return mProfileManager.getA2dpProfile().getConnectionStatus(mDevice) ==
-                BluetoothProfile.STATE_CONNECTED;
+        if (mProfileManager.getA2dpProfile() != null) {
+            return mProfileManager.getA2dpProfile().getConnectionStatus(mDevice) ==
+                    BluetoothProfile.STATE_CONNECTED;
+        } else if (mProfileManager.getA2dpSinkProfile() != null) {
+            return mProfileManager.getA2dpSinkProfile().getConnectionStatus(mDevice) ==
+                    BluetoothProfile.STATE_CONNECTED;
+        } else {
+            if (Utils.D)
+                Log.d(TAG, "isA2dpDevice: no A2DP source or sink profile found");
+
+            return false;
+        }
     }
 
     /**
-- 
2.17.1

